### PR TITLE
Fix row theme and column theme cutting off when entering into blank region of Grid

### PR DESCRIPTION
**Pull request by jassmith**
CleanShot 2024-01-23 at 22 33 13@2x

Previously everything below the last data row would just be white

### CODE REVIEW

File: packages/core/src/common/styles.ts [hunk 1]
```diff
@@ -159,7 +160,11 @@ export function mergeAndRealizeTheme(theme: Theme, ...overlays: Partial<Theme |
             for (const key in overlay) {
                 // eslint-disable-next-line no-prototype-builtins
                 if (overlay.hasOwnProperty(key)) {
-                    merged[key] = (overlay as any)[key];
+                    if (key === "bgCell") {
+                        merged[key] = blend(overlay[key] as string, merged[key]);
```
Comment (jassmith) [inline packages/core/src/common/styles.ts:164]: Without doing this we retain the previous requirement that bgCell always be a solid color. Now only the main theme need be a solid color.

File: packages/core/src/docs/examples/theme-per-row.stories.tsx [hunk 2]
```diff
@@ -35,27 +35,39 @@ export default {
...
-                          bgCell: "#f0f8ff",
-                          borderColor: "#3f90e0",
+                          bgCell: "#e0f0ff88",
+                          //   borderColor: "#3f90e0",
                       }
             }
             onCellEdited={setCellValue}
             onColumnResize={onColumnResize}
-            rows={1_000_000}
+            rows={10}
```
Comment (jassmith) [inline packages/core/src/docs/examples/theme-per-row.stories.tsx:70]: This makes it easier to see the change

File: packages/core/src/internal/data-grid/render/data-grid-render.lines.ts [hunk 3]
```diff
@@ -146,6 +146,134 @@ export function overdrawStickyBoundaries(
...
+        }
+    }
+
+    if (toDraw.length === 0) return;
+
+    let color: string | undefined;
+    ctx.beginPath();
+    // render in reverse order because we computed and added the columns last, but they should actually be lower
+    // priority than the rows.
+    for (let i = toDraw.length - 1; i >= 0; i--) {
```
Comment (jassmith) [inline packages/core/src/internal/data-grid/render/data-grid-render.lines.ts:258]: The complexity here is just to reduce the number of draw calls. Fewer draw calls === faster

**Comment by BrianHung** [comment]
image

Seems to be a missing row when trailingRowOptions and onRowAppended are defined.

**Review by BrianHung — CHANGES_REQUESTED** [review]

File: packages/core/src/internal/data-grid/render/data-grid-render.lines.ts [hunk 4]
```diff
@@ -146,6 +146,134 @@ export function overdrawStickyBoundaries(
...
+                    y: ty,
+                    w: maxX - minX,
+                    h: rh,
+                    color: rowThemeBgCell,
+                });
+            }
+        }
+
+        y += rh;
+        if (row < rows) y2 = y;
```
[the remark this answers is not in the document]
Comment (jassmith) [reply packages/core/src/internal/data-grid/render/data-grid-render.lines.ts:219]: Thank you! Great catch.

**Comment by BrianHung** [comment]
LGTM after that fix :)

**Pull request merged**